src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexMatchCollection.cs

```diff
@@ -134,22 +134,10 @@ bool ICollection.IsSynchronized
         /// </summary>
         void ICollection.CopyTo(Array array, int arrayIndex)
         {
-            if ((array != null) && (array.Rank != 1))
-            {
-                throw new ArgumentException(SR.Arg_RankMultiDimNotSupported);
-            }
+            if (!_done)
+                GetMatch(s_infinite);
```

Comment:
I would prefer either of the following to this change:

1. Keep the reference to `Count` (with the associated comment)
2. Refactor this conditional call to `GetMatch` to its own method (e.g. `LazyInit` or something), and call the new method from both the `Count` and this method